app/main/views/agreements.py

```diff
@@ -106,14 +106,19 @@ def update_framework_agreement(agreement_id):
     json_has_required_keys(json_payload, ["agreement"])
     update_json = json_payload["agreement"]
 
-    json_has_keys(update_json, optional_keys=['signedAgreementDetails', 'signedAgreementPath'])
+    json_has_keys(
+        update_json,
+        optional_keys=['signedAgreementDetails', 'signedAgreementPath', 'countersignedAgreementPath']
+    )
+
+    if (
+        framework_agreement.signed_agreement_returned_at
+        and ('signedAgreementDetails' in update_json or 'signedAgreementPath' in update_json)
+    ):
+        abort(400, "Can not update signedAgreementDetails or signedAgreementPath if agreement has been signed")
 
-    # TODO Behaviour to be introduced after next step of refactoring
```

Comment:
Are we ready for removing the comment yet? This will break current behaviour for people trying to upload a subsequent document.  Maybe hold off until we do the full "multiple FAs" piece next?